src/Language/PureScript/Bridge/Printer.hs

```diff
@@ -83,9 +83,9 @@ sumTypeToText st@(SumType t cs) = T.unlines $
     genericConstrainsInner = T.intercalate ", " $ map genericInstance sumTypeParameters
     stpLength = length sumTypeParameters
     bracketWrap x = "(" <> x <> ")"
-    sumTypeParameters = filter isTypeParameter $ getUsedTypes st
-    -- TODO: I find this stupid. Would much rather like some other way to make this check?
-    isTypeParameter tp = T.toLower (_typeName tp) == _typeName tp
+    sumTypeParameters = filter isTypeParam . Set.toList $ getUsedTypes st
+    -- NOTE: only this captures monadic type parameters. See test/Spec.hs
+    isTypeParam typ = T.toLower (_typeName typ) == _typeName typ
```

Comment:
Please see `test/Spec` for it. Solution that you have proposed doesn't cover this case:
```
data Bar2 a b m =
    Bar1 (Maybe a)
  | Bar2 (Either a b)
  | Bar3 a
  | Bar4 {
      myMonadicResult :: m b
    }

derive instance genericBar2 :: (Generic a, Generic b, Generic (m b)) => Generic (Bar2 a b m)  
```
If you have some other way to handle this case, please comment. I don't know how to properly handle monadic type parameters except this way, which I don't like